docs/modules/agents/toolkits/examples/spark_sql.ipynb

```diff
@@ -0,0 +1,342 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Spark SQL Agent\n",
+    "\n",
+    "This notebook shows how to use agents to interact with a Spark SQL. Similar to [SQL Database Agent](https://python.langchain.com/en/latest/modules/agents/toolkits/examples/sql_database.html), it is designed to address general inquiries about a database and facilitate error recovery.\n",
+    "\n",
+    "**NOTE: this agent calls the Python agent under the hood, which executes LLM generated Python code - this can be bad if the LLM generated Python code is harmful. Use cautiously.**"
```

Comment:
Sorry, but I can’t find out where is python agent used? It looks like this is using sql engine directly (+LLM). Do you mind sharing the code pointer where python agent is used? Thanks

Reply:
Nice catch. The sentence was copied form another notebook and it is a mistake. I just updated it.